wazo_router_confd/services/carrier.py

```diff
@@ -3,23 +3,40 @@
 
 from sqlalchemy.orm import Session
 
+from wazo_router_confd.auth import Principal
 from wazo_router_confd.models.carrier import Carrier
 from wazo_router_confd.schemas import carrier as schema
+from wazo_router_confd.services import tenant as tenant_service
 
 
-def get_carrier(db: Session, carrier_id: int) -> Carrier:
-    return db.query(Carrier).filter(Carrier.id == carrier_id).first()
+def get_carrier(db: Session, principal: Principal, carrier_id: int) -> Carrier:
+    db_carrier = db.query(Carrier).filter(Carrier.id == carrier_id)
+    if principal is not None and principal.tenant_uuid:
+        db_carrier = db_carrier.filter(Carrier.tenant_uuid == principal.tenant_uuid)
+    return db_carrier.first()
 
 
-def get_carrier_by_name(db: Session, name: str) -> Carrier:
-    return db.query(Carrier).filter(Carrier.name == name).first()
+def get_carrier_by_name(db: Session, principal: Principal, name: str) -> Carrier:
+    db_carrier = db.query(Carrier).filter(Carrier.name == name)
+    if principal is not None and principal.tenant_uuid:
+        db_carrier = db_carrier.filter(Carrier.tenant_uuid == principal.tenant_uuid)
+    return db_carrier.first()
 
 
-def get_carriers(db: Session, offset: int = 0, limit: int = 100) -> schema.CarrierList:
-    return schema.CarrierList(items=db.query(Carrier).offset(offset).limit(limit).all())
+def get_carriers(
+    db: Session, principal: Principal, offset: int = 0, limit: int = 100
+) -> schema.CarrierList:
+    items = db.query(Carrier)
+    if principal is not None and principal.tenant_uuid:
+        items = items.filter(Carrier.tenant_uuid == principal.tenant_uuid)
```

Comment:
The filter on `GET /resources` here is the good one. If we eventually need a list of all resources in a tenant and all of its subtenant we use the query parameter `recurse=true` on the HTTP requests but if it's not needed I suggest that we do not implement it.